third_party/cabal2bazel/bzl/cabal_package.bzl

```diff
@@ -141,13 +144,34 @@ def _get_build_attrs(
           src = f,
           out = module_map[m],
       )
+
+    for f,m,out in _glob_modules(d, ".chs", ".chs"):
+      chs_rule = name + "-" + m
+      symlink_rule = name + "-" + m + "-symlink"
+      hazel_symlink(
+          name = symlink_rule,
+          src = f,
+          out = srcs_dir + out,
+      )
+      # TODO: this will not work if one .chs file imports another.
+      # To mimic Cabal's behavior we should assume the module names are
+      # topologically ordered; i.e., make each one depend on all previous
+      # modules.  However, that's not easy the way the code is currently
+      # structured.
```

Comment:
Why is that not easy? Can't we collect names of previously created `c2hs_library` rules and pass them in `deps`?

Reply:
Since the code iterates over the output of `_glob_modules`, it is iterating over the `.chs` files, and creating the c2hs_library rules, in (I believe) lexical order of filename.

To wire up the dependencies the way you're suggesting, we'd need to change the code to create them in the order that the modules were listed in the Cabal file.  (Honestly, that would probably be a better structure for this code anyway, it's just more effort to get right...) .